### PR TITLE
feat(#648): add --disable-users command for merge-contacts action

### PR DESCRIPTION
**Pull request by kennsippell**
# Description

When `--disable-users` option is present during `merge-contacts`

1. Look for deleted docs in `json_docs` folder during upload-docs action
2. Look for users having facility_id which matches the deleted doc id
3. Removes the id from the user's list of facility_ids
4. Updates the user as appropriate (deletes it if no facility_ids remain)

#648 
Base branch is #647

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or integration tests where appropriate
- Backwards compatible: Works with existing data and configuration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: src/fn/upload-docs.js
Comment (sugat009) [inline src/fn/upload-docs.js:134]:
```suggestion
  warnAndPrompt(`This operation will update places for ${affectedUsers.length} user accounts or delete them: ${usernames}. Are you sure you want to continue?`);
```
Issue (non-blocking): The operation that proceeds with this check is to update or delete users. The prompt message seems out of place.

File: src/fn/upload-docs.js
Comment (sugat009) [inline src/fn/upload-docs.js:139]: suggestion: if `deletedDocIds` are sure to be `facilityIds` then, renaming it to `facilityIds` makes more sense

File: src/fn/upload-docs.js
Comment (sugat009) [inline src/fn/upload-docs.js:103]: issue: this function might be better suited for `src/lib/user-prompt.js`

File: src/fn/upload-docs.js
Comment (sugat009) [inline src/fn/upload-docs.js:142]: question: I noticed these fields might be redundant. Could you confirm if they're essential for the update endpoint or if they can be removed?

File: test/fn/upload-docs.spec.js
Comment (sugat009) [inline test/fn/upload-docs.spec.js:199]:
```suggestion
    it('one user disabled and one updated when single place has multiple users', async () => {
```

**Review by sugat009 — CHANGES_REQUESTED** [review]

File: src/fn/upload-docs.js
Comment (kennsippell) [reply src/fn/upload-docs.js:139]: sadly, at this point we can't guarantee they are facilities. app devs write many types of scripts which create documents and dump them into the `json_docs` folder to be uploaded by the `upload-docs` action.  

i don't think the paramater should be specific to facilities at this point... this function is kinda _checking if the deleted doc ids end up being user's facility_ids (affectedUsers)_

File: src/fn/upload-docs.js
Comment (kennsippell) [reply src/fn/upload-docs.js:142]: nice catch. i think you're right. i'll test through some scenarios

**Comment by kennsippell** [comment]
@sugat009 I'm really sorry about this, but I just realised that the difference between this and #652 would technically be a breaking change because of an interface change.  I need to bring some code from that change forward into this pull request.  

I think this work stands alone now with a seamless change into #652. It is also possible that it might just make sense to review #652 against main. I'll let you decide how you want to proceed! 

So the changes are:

1. The `--disable-users` is a parameter which is passed to the `merge-contacts` action now, rather than being a parameter which is passed to `upload-docs` action
2. Assert if CHT Core version is less than 4.7

File: src/fn/upload-docs.js
Comment (kennsippell) [reply src/fn/upload-docs.js:137]: did something similar

File: src/fn/upload-docs.js
Comment (sugat009) [inline src/fn/upload-docs.js:137]: question: I've noticed an inconsistency in the command structure. The `disable-users` argument was moved from the `disable-users` command to the `merge-contacts` command. However, after reviewing the `merge-contacts` command, I don't see a direct invocation of `upload-docs`. 

Are we looking at a scenario where these commands should be combined, such as `cht merge-contacts upload-docs --disable-users`? 

My concern is that running this against a CHT 4.6.0 instance might result in an error, even if the `--disable-users` flag isn't explicitly passed.

Comment (kennsippell) [reply src/fn/upload-docs.js:139]: > My concern is that running this against a CHT 4.6.0 instance might result in an error, even if the --disable-users flag isn't explicitly passed.

You're right. Sorry that was a bit sloppy. I guess we want the `--disable-users` gate on both actions.

Comment (sugat009) [reply src/fn/upload-docs.js:141]: Yes, option 1 is the best option. Let's go with that.

Comment (kennsippell) [reply src/fn/upload-docs.js:141]: OK. That is what is in this PR now.

**Review by sugat009 — APPROVED** [review]

**Pull request merged**
